pygrocy/grocy.py

```diff
@@ -148,6 +153,27 @@ def consume_product(
             product_id, amount, spoiled, transaction_type
         )
 
+    def add_product_by_barcode(
+        self,
+        barcode: str,
+        amount: float,
+        price: float,
+        best_before_date: datetime = None
+    ):
```

Comment:
The Grocy API returns a `StockLogEntry` for this - the existing consume/add methods simply return the `StockLogEntry` as the dictionary which results from calling `requests.json()`. I have defined `StockLogEntry` as a class so I will map them to this?

Reply:
The API request returns a `StockLogObject` - the existing add/consume methods simply return the result of calling `requests.json()` on the `StockLogObject`. I have created a `StockLogObject` class and created the mapping to `Product` - it's a bit of a weird one because the only real usable data you get back in the `StockLogObject` is the product ID - everything else just relates to the stock entry that has just been created - so it has to immediately call `Product.getDetails()` to populate the rest of the object.